### ISSUE TITLE
Make playpen compile with debug asserts/overflow checks per default.

### ISSUE THREAD

**Issue by Kimundi** [rust-lang/rust-playpen#88]
Eg, so that this panics:
```
<kimundi> playbot: let x = 5u8; x - 10
-playbot/#rust- 251
```

**Comment by mkpankov** [comment]
I'd rather say there should be another switch - as for optimization (`-O0` ... `-O2`). Makes it easy to identify the current mode and takes no assumptions on what the default is.

Also, allows to argue about features which behave differently in debug and release modes, like @Kimundi 's example :p

**Comment by Kimundi** [comment]
I just realized that overflow checks are enabled with `-O0`, so maybe this is unneeded after all.

**Comment by mkpankov** [comment]
I couldn't trigger compile-time check with `-O0`: http://imgur.com/zxrZ59z. Besides, optimization levels and debugging capabilities are orthogonal.

**Comment by alexcrichton** [comment]
Yeah I've often thought that opt-by-default may be the wrong default here as well, although I think we have two options here:

* Add a separate toggle for debug assertions
* Switch to O0 by default

If we add a separate toggle it does seem prudent to enable them by default!

**Comment by Kimundi** [comment]
@mkpankov: Your screenshot shows a compiletime check though.

However, this issue is about the debug _runtime_ checks.

**Comment by mkpankov** [comment]
@Kimundi right, I misunderstood what "application terminated with error code 101" means. Turns out this line is printed on compiler errors and runtime errors the same.
